### PR TITLE
document the format() function

### LINKED ISSUE #1308: Document: refactor(expr): support variadic function in `#[function]` macro
This issue tracks the documentation update needed for the merged PR #12178.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/12178
Source PR Merged At: 2023-09-15T09:21:11Z

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

Reformatted the string functions (from table cells to sections with headings), and documented the format() function.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #1308

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-string.md [hunk 1]
```diff
@@ -17,46 +16,563 @@ title: String functions and operators
...
+%type
+```
+
+*type* is the type of format conversion to use to generate the output of the format specifier.
+
+The allowed values for *type* are:
+
+- `s`: Formats the argument value as a string. NULL is treated as an empty string.
+- `I`: Treats the argument value as an SQL identifer.
+- `L`: Quotes the argument value as an SQL literal.
```
Comment (wangrunji0408) [inline docs/sql/functions-operators/sql-function-string.md]: `L` is not supported for now. We can hide it temporarily.

**Review by wangrunji0408 — COMMENTED** [review]
Other part of the `format` function LGTM!

File: docs/sql/functions-operators/sql-function-string.md [hunk 2]
```diff
@@ -17,46 +16,563 @@ title: String functions and operators
...
+
+### `format`
+
+This function produces output formatted according to a format string, in a style similar to the C function `sprintf`.
+
+```bash title=Syntax
+format( format_string [, format_arg [, ...] ] ) → output_string
+```
+
+The *format_string* specifies how the *output_string* should be formatted. It consists of text and format specifiers. Text is copied directly to the output string. Format specifies are placeholders for the arguments to be inserted into the output string.
```
Comment (emile-00) [inline docs/sql/functions-operators/sql-function-string.md]: specifiers

File: docs/sql/functions-operators/sql-function-string.md [hunk 3]
```diff
@@ -17,46 +16,563 @@ title: String functions and operators
...
+```
+
+---
+
+### `format`
+
+This function produces output formatted according to a format string, in a style similar to the C function `sprintf`.
+
+```bash title=Syntax
+format( format_string [, format_arg [, ...] ] ) → output_string
```
Comment (emile-00) [inline docs/sql/functions-operators/sql-function-string.md:154]: Should we add a note reminding users that the number of format specifiers should be equal to or less than the number of arguments?

**Pull request merged**